cmd/main.go

```diff
@@ -261,7 +254,9 @@ func InitShouldIgnoreArg(arg string) error {
 	return fmt.Errorf("unknown argument for --ignore-on-exit: %s\nvalid arguments:\n  %s", arg, strings.Join(validArgs, "\n  "))
 }
 
-// ShowError returns true if should show error, otherwise returns false
 func ShowError(kind string) bool {
-	return strings.EqualFold(ignoreOnExitVar, "none") || (!strings.EqualFold(ignoreOnExitVar, "all") && !strings.EqualFold(ignoreOnExitVar, kind))
+	if strings.EqualFold(ignoreOnExitVar, "none") || strings.EqualFold(ignoreOnExitVar, "results") {
+		return true
+	}
+	return false
 }
```

Comment:
I think you forgot to check the `kind`

Reply:
Yes, you are right. I just fixed it but noticed another issue so will work on it later today/tomorrow and keep you updated.

Reply:
**Proposed Changes:**
- Simplified and modified showError. Added a global variable (showErrorKind) to determine the behavior of the function.

**Next to do:**
- Remove log.Fatal messages from rule.go file